core/commands/storage/upload/upload/init.go

```diff
@@ -47,27 +63,39 @@ the shard and replies back to client for the next challenge step.`,
 		cmds.StringArg("storage-length", true, false, "Store file for certain length in days."),
 		cmds.StringArg("shard-size", true, false, "Size of each shard received in bytes."),
 		cmds.StringArg("shard-index", true, false, "Index of shard within the encoding scheme."),
+		cmds.StringArg("agree-reject", true, false, "Host agrees or rejects the extended period."),
```

Comment:
There are two issues here:
1) It breaks compatibility since you shifted the last "upload-peer-id" position, all sign-related calls (go-btfs-api and js-btfs-api need to be updated - a lot of work).
2) The policy should be determined by host. If you pass it from renter then it is determined by renter, which is not right?